### PR TITLE
ipam and vlan specializers to controller/pkg/reconcilers

### PR DESCRIPTION
**Pull request by ssharmasandeep**

### CODE REVIEW

**Comment by ssharmasandeep** [comment]
After this is merged I will create the concluding PR with the operator/nephio-controller-manager main.go final changes with proper config passed in the loop for setupwithmanager calls.
-Sandeep

**Comment by ssharmasandeep** [comment]
There is some duplicate code now in controller/pkg/specializer-reconciler, will wait for @henderiw review to handle that.

File: controllers/pkg/reconcilers/vlan-specializer/reconciler.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+)
+
+func init() {
+	reconcilerinterface.Register("vlamreconciler", &reconciler{})
```
Comment (henderiw) [inline controllers/pkg/reconcilers/vlan-specializer/reconciler.go:48]: call it vlanspecializer

File: controllers/pkg/reconcilers/ipam-specializer/reconciler.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+)
+
+func init() {
+	reconcilerinterface.Register("ipamreconciler", &reconciler{})
```
Comment (henderiw) [inline controllers/pkg/reconcilers/ipam-specializer/reconciler.go:49]: call it ipamspecializer

**Comment by henderiw** [comment]
move condition to pkg/porch/condition

File: controllers/pkg/reconcilers/vlan-specializer/reconciler.go (hunk 1 above)
Comment (ssharmasandeep) [reply controllers/pkg/reconcilers/vlan-specializer/reconciler.go]: Done

File: controllers/pkg/reconcilers/ipam-specializer/reconciler.go (hunk 2 above)
Comment (ssharmasandeep) [reply controllers/pkg/reconcilers/ipam-specializer/reconciler.go]: Done

**Pull request merged**